pkg/services/oracle/jsonpath/jsonpath.go

```diff
@@ -0,0 +1,476 @@
+package jsonpath
+
+import (
+	"strconv"
+	"strings"
+
+	json "github.com/virtuald/go-ordered-json"
+)
+
+type (
+	// pathTokenType represents single JSONPath token.
+	pathTokenType byte
+
+	// pathParser combines JSONPath and a position to start parsing from.
+	pathParser struct {
+		s     string
+		i     int
+		depth int
+	}
+)
+
+const (
+	pathRoot pathTokenType = iota
+	pathDot
+	pathLeftBracket
+	pathRightBracket
+	pathAsterisk
+	pathComma
+	pathColon
+	pathIdentifier
+	pathString
+	pathNumber
+)
+
+const maxNestingDepth = 6
+
+// Get returns substructures of value selected by path.
+// The result is always non-nil unless path is invalid.
+func Get(path string, value interface{}) ([]interface{}, bool) {
+	if path == "" {
+		return []interface{}{value}, true
+	}
+
+	p := pathParser{
+		depth: maxNestingDepth,
+		s:     path,
+	}
+
+	typ, _, ok := p.nextToken()
+	if !ok {
+		return nil, false
+	}
+
+	if typ != pathRoot {
+		return nil, false
+	}
+
+	objs := []interface{}{value}
+	for p.i < len(p.s) {
+		typ, _, ok := p.nextToken()
+		if !ok {
+			return nil, false
+		}
+
+		switch typ {
+		case pathDot:
+			objs, ok = p.processDot(objs)
+		case pathLeftBracket:
+			objs, ok = p.processLeftBracket(objs)
+		default:
+			return nil, false
+		}
+
+		if !ok {
+			return nil, false
+		}
+	}
+
+	if objs == nil {
+		objs = []interface{}{}
+	}
+	return objs, true
+}
+
+func (p *pathParser) nextToken() (pathTokenType, string, bool) {
```

Comment:
And then return `pathTokenType, string` here saving on `bool` and some error checks.